api/checks/azure_test.go

```diff
@@ -0,0 +1,118 @@
+// +build small
+
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package checks
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	logrustest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"github.com/web-platform-tests/wpt.fyi/shared/sharedtest"
+)
+
+func TestHandleAzurePipelinesEvent(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	sha := strings.Repeat("0123456789", 4)
+	detailsURL := "https://dev.azure.com/web-platform-tests/b14026b4-9423-4454-858f-bf76cf6d1faa/_build/results?buildId=123"
+	event := getCheckRunCreatedEvent("completed", "lukebjerring", sha)
+	event.CheckRun.DetailsURL = &detailsURL
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
@mdittmer - here's a "fake" http server (as opposed to a mock one). The calling code just expects "given this http.Client, and this URL, fetch it" so we mock the URL we give out (as the server's URL) and the client we give out (as the server's Client).